pkg/service/customer.go

```diff
@@ -44,17 +57,42 @@ func (c *Customer) Start(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("falha ao iniciar o servidor GRPC: %w", err)
 	}
-	if err = c.grpc.Start(ctx); err != nil {
-		return fmt.Errorf("falha ao inicia o servidor GRPC: %w", err)
-	}
+	go func() {
+		if grpcErr := c.grpc.Start(ctx); grpcErr != nil && !errors.Is(grpcErr, grpc.ErrServerStopped) {
+			c.cfg.Logger.Error("falha ao iniciar o servidor GRPC: %w", zap.Error(grpcErr))
+			c.asyncErrorChannel <- grpcErr
+		}
+	}()
 
 	c.srv, err = server.NewHTTP(c.cfg, ch)
 	if err != nil {
 		return fmt.Errorf("falha ao iniciar o servidor HTTP: %w", err)
 	}
 
-	if err = c.srv.Start(ctx); err != nil {
-		return fmt.Errorf("falha ao iniciar o servidor HTTP: %w", err)
+	go func() {
+		if httpErr := c.srv.Start(ctx); httpErr != nil && !errors.Is(httpErr, http.ErrServerClosed) {
+			c.cfg.Logger.Error("falha ao iniciar o servidor HTTP: %w", zap.Error(httpErr))
```

Comment:
```suggestion
			c.cfg.Logger.Error("falha ao iniciar o servidor HTTP", zap.Error(httpErr))
```

O zap vai colocar o campo de erro na mensagem, nao vai concatenar a mensagem fixa com o erro.